File: solutions/csharp/log-levels/LogLevels.cs
using Microsoft.VisualBasic;

static class LogLine
{
    public static string Message(string logLine)
    {
        if (logLine[1] == 'I')
            return logLine[7..].Trim();
        else if (logLine[1] == 'W')
            return logLine[10..].Trim();
        else
            return logLine[8..].Trim();
    }

    public static string LogLevel(string logLine)
    {
        if (logLine[1] == 'I')
            return "info";
        else if (logLine[1] == 'W')
            return "warning";
        else
            return "error";
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine) + " (" + LogLevel(logLine) + ")";
    }
}

File: solutions/cpp/log-levels/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    if (line.substr(0, 7) == "[INFO]:")
        return line.substr(8);
    else if (line.substr(0, 10) == "[WARNING]:")
        return line.substr(11);
    else if (line.substr(0, 8) == "[ERROR]:")
        return line.substr(9);
    else
        return "";
}

std::string log_level(std::string line) {
    if (line.substr(0, 7) == "[INFO]:")
        return line.substr(1, 4);
    else if (line.substr(0, 10) == "[WARNING]:")
        return line.substr(1, 7);
    else if (line.substr(0, 8) == "[ERROR]:")
        return line.substr(1, 5);
    else
        return "";
}

std::string reformat(std::string line) {
    return message(line) + " (" + log_level(line) + ")";
}

}  // namespace log_line
